lib/spack/spack/solver/asp.py

```diff
@@ -3999,6 +4007,31 @@ def _specs_from_mirror():
         return []
 
 
+def _specs_from_environment(path):
+    active_env = ev.active_environment()
+
+    if path:
+        if not ev.is_env_dir(path) and ev.exists(path):
+            path = ev.root(path)
+
+    if active_env and path in active_env.included_specs_by_hash:
+
+        def _specs_for_env():
+            for s in active_env.included_specs_by_hash[path].values():
+                yield s
+
+        return _specs_for_env()
+    elif path:
+        env = ev.read(path)
+    else:
+        env = active_env
```

Comment:
I think this behavior is incorrect -- I thought if we're just reusing from "environment" that meant that we're reusing all concrete specs from included environments regardless of path.

Reply:
yes, that is what this should do. The path is only if you want to select the subset of concrete specs from an environment, but if the path is not provided it will use all of the `concretized_specs`, which includes all included concrete specs.